chainntnfs/txconfnotifier.go

```diff
@@ -98,78 +101,116 @@ func NewTxConfNotifier(startHeight uint32, reorgSafetyLimit uint32) *TxConfNotif
 
 // Register handles a new notification request. The client will be notified when
 // the transaction gets a sufficient number of confirmations on the blockchain.
-// If the transaction has already been included in a block on the chain, the
-// confirmation details must be given as the txConf argument, otherwise it
-// should be nil. If the transaction already has the sufficient number of
-// confirmations, this dispatches the notification immediately.
-func (tcn *TxConfNotifier) Register(ntfn *ConfNtfn, txConf *TxConfirmation) error {
+//
+// NOTE: If the transaction has already been included in a block on the chain,
+// the confirmation details must be provided with the UpdateConfDetails method,
+// otherwise we will wait for the transaction to confirm even though it already
+// has.
+func (tcn *TxConfNotifier) Register(ntfn *ConfNtfn) error {
```

Comment:
Commit message says "registrations for unconfirmed transactions only", but seems like this will also be called fro already confirmed transactions?

Reply:
Clarified things a bit, let me know what you think.